quark.c

```diff
@@ -408,43 +479,179 @@ process_cache_inherit(struct quark_queue *qq, struct quark_process *qp, int ppid
 	}
 }
 
+
 static void
 process_cache_delete(struct quark_queue *qq, struct quark_process *qp)
 {
+	struct gc_link	*gc;
+
+	gc = &qp->gc;
 	RB_REMOVE(process_by_pid, &qq->process_by_pid, qp);
-	if (qp->gc_time)
-		TAILQ_REMOVE(&qq->event_gc, qp, entry_gc);
+	gc_unlink(qq, gc);
 	free(qp);
 }
 
 static int
-process_cache_gc(struct quark_queue *qq)
+process_by_pid_cmp(struct quark_process *a, struct quark_process *b)
 {
-	struct quark_process	*qp;
-	u64			 now;
-	int			 n;
+	if (a->pid < b->pid)
+		return (-1);
+	else if (a->pid > b->pid)
+		return (1);
 
-	now = now64();
-	n = 0;
-	while ((qp = TAILQ_FIRST(&qq->event_gc)) != NULL) {
-		if (AGE(qp->gc_time, now) < qq->cache_grace_time)
-			break;
-		process_cache_delete(qq, qp);
-		n++;
+	return (0);
+}
+
+/*
+ * Socket stuff
+ */
+
+#if 0
+static int
+sockaddr_to_str(const struct quark_sockaddr *qsa, char *buf, size_t len)
+{
+	size_t	n;
+	int	r;
+
+	if (inet_ntop(qsa->af, qsa->addr6, buf, len) == NULL)
+		return (-1);
+	n = strlen(buf);
+	buf += n;
+	len -= n;
+	if (len == 0)
+		return (-1);
+	r = snprintf(buf, len, ":%d", ntohs(qsa->port));
+	if (r <= 0 || r >= (int)len)
+		return (-1);
+
+	return (0);
+}
+
+static int
+socket_to_str(const struct quark_socket *qsk, char *buf, size_t len)
+{
+	size_t	n;
+
+	if (sockaddr_to_str(&qsk->local, buf, len) == -1)
+		return (-1);
+	n = strlcat(buf, " -> ", len);
+	if (n >= len)
+		return (-1);
+	buf += n;
+	len -= n;
+	if (sockaddr_to_str(&qsk->remote, buf, len) == -1)
+		return (-1);
+
+	return (0);
+}
+#endif
+
+static struct quark_socket *
+socket_cache_get(struct quark_queue *qq,
+    struct quark_sockaddr *local, struct quark_sockaddr *remote, int alloc)
```

Comment:
can local and remote be const?

Reply:
Could, as it's not an external facing function, I wouldn't care much (most of the other code is node const)